pkg/resource/feature_group/manager_test_suite_test.go

```diff
@@ -0,0 +1,196 @@
+	}
+	//TODO: use reflection, template to auto generate this block/method.
+	switch apiName {
+	case "CreateFeatureGroup":
+		var output svcsdk.CreateFeatureGroupOutput
+		return &output, nil
+	case "CreateFeatureGroupRequest":
+		var output svcsdk.CreateFeatureGroupOutput
+		return &output, nil
+	case "CreateFeatureGroupWithContext":
+		var output svcsdk.CreateFeatureGroupOutput
+		return &output, nil
+	case "DeleteFeatureGroup":
+		var output svcsdk.DeleteFeatureGroupOutput
+		return &output, nil
+	case "DeleteFeatureGroupRequest":
+		var output svcsdk.DeleteFeatureGroupOutput
+		return &output, nil
+	case "DeleteFeatureGroupWithContext":
+		var output svcsdk.DeleteFeatureGroupOutput
+		return &output, nil
+	case "DescribeFeatureGroup":
+		var output svcsdk.DescribeFeatureGroupOutput
+		return &output, nil
+	case "DescribeFeatureGroupRequest":
+		var output svcsdk.DescribeFeatureGroupOutput
+		return &output, nil
+	case "DescribeFeatureGroupWithContext":
+		var output svcsdk.DescribeFeatureGroupOutput
+		return &output, nil
+	case "ListFeatureGroups":
+		var output svcsdk.ListFeatureGroupsOutput
+		return &output, nil
+	case "ListFeatureGroupsPages":
+		var output svcsdk.ListFeatureGroupsOutput
+		return &output, nil
+	case "ListFeatureGroupsRequest":
+		var output svcsdk.ListFeatureGroupsOutput
+		return &output, nil
+	case "ListFeatureGroupsWithContext":
+		var output svcsdk.ListFeatureGroupsOutput
+		return &output, nil
+	}
+	return nil, errors.New(fmt.Sprintf("no matching API name found for: %s", apiName))
+}
+
+func (d *testRunnerDelegate) Equal(a acktypes.AWSResource, b acktypes.AWSResource) bool {
+	ac := a.(*resource)
+	bc := b.(*resource)
+	// Ignore LastTransitionTime since it gets updated each run.
+	opts := []cmp.Option{cmpopts.EquateEmpty(), cmpopts.IgnoreFields(ackv1alpha1.Condition{}, "LastTransitionTime")}
+
+	if cmp.Equal(ac.ko.Status, bc.ko.Status, opts...) {
+		return true
+	} else {
+		fmt.Printf("Difference (-expected +actual):\n\n")
+		fmt.Println(cmp.Diff(ac.ko.Status, bc.ko.Status, opts...))
+		return false
+	}
+}
+
+// Checks to see if the given yaml file, with name stored as expectation,
+// matches the yaml marshal of the AWSResource stored as actual.
+func (d *testRunnerDelegate) YamlEqual(expectation string, actual acktypes.AWSResource) bool {
+     	// Get the file name of the expected yaml.
+	expectedYamlFileName := TestDataDirectory + "/" + expectation
+
+	// Build a tmp file for the actual yaml.
+	actualResource := actual.(*resource)
+	actualYamlByteArray, _ := yaml.Marshal(actualResource.ko)
+	actualYamlFileName := buildTmpFile("actualYaml", actualYamlByteArray)
+	defer os.Remove(actualYamlFileName)
+	if "" == actualYamlFileName {
+	  fmt.Printf("Could not create temporary actual file.\n")
+	  return false
+	}
+
+	// Replace Timestamps that would show up as different.
+	_, err := exec.Command("sed", "-r", "-i", ReplaceTimestampRegExp, actualYamlFileName).Output()
```

Comment:
Nit: 
I feel like it would be better to just do string manipulation before creating the temp file on line 138

Reply:
Given that we are doing a line by line diff as an exec.Command, the sed matches very nicely with that intent which is line by line comparison, and appears to me to be clearer to the reader.

It would also be more code and a little less clear to iterate through the fields of the yaml object (checking their types).
Unless is there a really easy way to iterate through a yaml object checking the type of each field and then doing a replace.